### DISCUSSION TITLE
Using SPI/I2C/CAN to communicate with another microcontroller

### DISCUSSION THREAD

**Discussion by Pr0gmr** [Sevenstax/FreeV2G#118] [Ideas]
Hi,

Is it possible to use SPI, I2C or CAN to communicate with another micro-controller to transfer information and not to just use it instead of the ethernet to initialize and run the given code? 

If it is possible can you direct to a sample code that establishes the communication.
Thank you

**Comment by jpo-stx** [comment]
Hi @Pr0gmr ,
I am not really sure what you mean.
For the host controller interface (HCI) of the Whitebeet, there is only SPI and Ethernet available for controlling the Whitebeet.
Do you mean that you have a host controller connected to the Whitebeet and want to add another controller?

**Reply by Pr0gmr** [reply]
Hey @jpo-stx ,
For example if I want to read the EV charging requirements through the EVSE onto another micro-controller, can I do it through SPI/I2C/CAN?
Basically, the requirement is, the EV and EVSE is connected through the CP and PE while another micro-controller connects to EVSE through SPI/I2C/CAN and it reads the EV charging details/requirements.

**Reply by jpo-stx** [reply]
This is exactly what the Whitebeet is for. You can use the SPI interface if you do not have ethernet on your controller.
You will receive the charging parameters from the EV (DC target current/voltage, state of charge) and send back your parameters (DC present current/voltage).
